### ISSUE TITLE
Feature deprecated in h5py is used in TF1.5

### ISSUE THREAD

**Issue by feranick** [tensorflow/tensorflow#16587]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux, OS X
- **TensorFlow installed from (source or binary)**: source and binary
- **TensorFlow version (use command below)**: 1.5
- **Python version**: 3.6.4
- **Bazel version (if compiling from source)**: 0.9
- **GCC/Compiler version (if compiling from source)**: 
- **CUDA/cuDNN version**: 9.0 - 7.0
- **GPU model and memory**: GTX1060, GTX 1050Ti 
- **Exact command to reproduce**:
`sudo pip3 install h5py`
run python3, from there, type:
`import tensorflow as tf`

### Describe the problem
A feature of h5py used in TF 1.5 is deprecated, in particular: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated.

### Source code / logs
Warning message: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters

**Comment by imsheridan** [comment]
Created a PR #16591 to fix the FutureWarning.

**Comment by reedwm** [comment]
Marking as contributions welcome since there is a PR.

**Comment by cancan101** [comment]
per https://github.com/tensorflow/tensorflow/pull/16591#issuecomment-371191027, this still seems to be an issue

**Comment by imsheridan** [comment]
It seems I cannot repro this issue with TF1.6 on my side, @cancan101 could you pls help confirm your detailed info as below:
#### OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
#### TensorFlow installed from (source or binary): 
#### TensorFlow version (use command below):
#### Python version: 
#### Bazel version (if compiling from source):
#### GCC/Compiler version (if compiling from source):

**Comment by cancan101** [comment]
Running `tf_env_collect.sh`:

```
== cat /etc/issue ===============================================
Darwin Alexs-MacBook-Pro-5.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
Mac OS X 10.12.6

== are we in docker =============================================
No

== compiler =====================================================
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

== uname -a =====================================================
Darwin Alexs-MacBook-Pro-5.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64

== check pips ===================================================
numpy (1.14.1)
protobuf (3.5.2)
tensorflow (1.6.0)

== check for virtualenv =========================================
True

== tensorflow import ============================================
tf.VERSION = 1.6.0
tf.GIT_VERSION = v1.6.0-0-gd2e24b6039
tf.COMPILER_VERSION = v1.6.0-0-gd2e24b6039
Sanity check: array([1], dtype=int32)
/Users/alex/.virtualenvs/test3/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: compiletime version 3.6 of module 'tensorflow.python.framework.fast_tensor_util' does not match runtime version 3.5
  return f(*args, **kwds)
/Users/alex/.virtualenvs/test3/lib/python3.5/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters

== env ==========================================================
LD_LIBRARY_PATH is unset
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
tf_env_collect.sh: line 105: nvidia-smi: command not found

== cuda libs  ===================================================
```

**Comment by kylemcdonald** [comment]
i just compiled from source in hopes this was fixed, but alas it is not. in addition to the information below, my h5py version is 2.7.1:

```
== uname -a =====================================================
Linux iyoiyo-gpu 4.13.0-38-generic #43~16.04.1-Ubuntu SMP Wed Mar 14 17:48:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
numpy (1.14.2)
numpydoc (0.7.0)
protobuf (3.5.2.post1)
tensorflow (1.7.0)
tensorflow-gpu (1.7.0)
tensorflow-tensorboard (0.1.6)

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.7.0
tf.GIT_VERSION = b'v1.7.0-1568-gf9c5e71'
tf.COMPILER_VERSION = b'v1.7.0-1568-gf9c5e71'
Sanity check: array([1], dtype=int32)
/home/kyle/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters

== env ==========================================================
LD_LIBRARY_PATH /home/kyle/torch/install/lib:/usr/local/cuda/lib64:
DYLD_LIBRARY_PATH /home/kyle/torch/install/lib:/home/kyle/torch/install/lib:/home/kyle/torch/install/lib:

== nvidia-smi ===================================================
Wed Apr 11 12:16:46 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 390.48                 Driver Version: 390.48                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:05:00.0  On |                  N/A |
|  0%   55C    P0    76W / 280W |    190MiB / 11175MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  Off  | 00000000:09:00.0 Off |                  N/A |
|  0%   52C    P0    74W / 280W |      2MiB / 11178MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1103      G   /usr/lib/xorg/Xorg                           146MiB |
|    0      2225      G   compiz                                        41MiB |
+-----------------------------------------------------------------------------+

== cuda libs  ===================================================
/usr/local/cuda-9.1/doc/man/man7/libcudart.7
/usr/local/cuda-9.1/doc/man/man7/libcudart.so.7
/usr/local/cuda-9.1/lib64/libcudart.so.9.1.85
/usr/local/cuda-9.1/lib64/libcudart_static.a
/usr/local/cuda-9.0/doc/man/man7/libcudart.7
/usr/local/cuda-9.0/doc/man/man7/libcudart.so.7
/usr/local/cuda-9.0/lib64/libcudart.so.9.0.176
/usr/local/cuda-9.0/lib64/libcudart_static.a
```

**Comment by cancan101** [comment]
you need to use the new rc for h5py.

**Comment by kylemcdonald** [comment]
i couldn't get it to build with the latest h5py. it might be because i didn't run `bazel clean` in between builds, so i am trying again now.

```
$ pip install git+https://github.com/h5py/h5py.git@2.8.0rc1`
$ ./configure
$ bazel build --config=opt --config=cuda --config=mkl //tensorflow/tools/pip_package:build_pip_package --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0"
bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tensorflow_pkg
...
ERROR: /home/kyle/Documents/setup/tensorflow/tensorflow/tools/api/generator/BUILD:27:1: Executing genrule //tensorflow/tools/api/generator:python_api_gen failed (Aborted): bash failed: error executing com
mand /bin/bash -c ... (remaining 1 argument(s) skipped)
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.8.16, library is 1.8.17
            SUMMARY OF THE HDF5 CONFIGURATION 
            ================================= 

General Information:
-------------------
                   HDF5 Version: 1.8.17
                  Configured on: Sat May 27 16:16:57 UTC 2017
                  Configured by: root@de9f433bbe70
                 Configure mode: production   
                    Host system: x86_64-unknown-linux-gnu
              Uname information: Linux de9f433bbe70 3.13.0-119-generic #166-Ubuntu SMP Wed May 3 12:18:55 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
                       Byte sex: little-endian
                      Libraries: static, shared
             Installation point: /home/kyle/anaconda3

Compiling Options:
------------------
               Compilation Mode: production   
                     C Compiler: /opt/rh/devtoolset-2/root/usr/bin/gcc ( gcc (GCC) 4.8.2 20140120 )
                         CFLAGS:  -m64
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-pro
totypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wun
reachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsaf
e-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wdouble-pro
motion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3
                      AM_CFLAGS:
                       CPPFLAGS:
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS:  -I/home/kyle/anaconda3/include
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS:
                     H5_LDFLAGS:
                     AM_LDFLAGS:  -L/home/kyle/anaconda3/lib
                Extra libraries: -lrt -lpthread -lz -ldl -lm
                       Archiver: ar
                         Ranlib: ranlib
              Debugged Packages:
                    API Tracing: no
Languages:
----------
                        Fortran: yes
               Fortran Compiler: /opt/rh/devtoolset-2/root/usr/bin/gfortran ( GNU Fortran (GCC) 4.8.2 20140120 )
          Fortran 2003 Compiler: yes
                  Fortran Flags:
               H5 Fortran Flags:
               AM Fortran Flags:
         Shared Fortran Library: yes
         Static Fortran Library: yes

                            C++: yes
                   C++ Compiler: /opt/rh/devtoolset-2/root/usr/bin/g++ ( g++ (GCC) 4.8.2 20140120 )
                      C++ Flags:  -DBOOST_MATH_DISABLE_FLOAT128 -m64
                   H5 C++ Flags:
                   AM C++ Flags:
             Shared C++ Library: yes
             Static C++ Library: yes

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: yes
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib)
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
Bye...
/bin/bash: line 1:  5901 Aborted                 (core dumped) 
...
INFO: Elapsed time: 2863.064s, Critical Path: 176.83s
FAILED: Build did NOT complete successfully   
```

**Comment by mohantym** [comment]
Hi @feranick ! 1.x versions are not supported anymore . Can you move this to closed status as it is not replicating in [2.8 version](https://colab.sandbox.google.com/gist/mohantym/7e50b763fc3bbc5fe46ee21f74a6ee63/github_16587.ipynb) ? Thanks!

**Comment by feranick** [comment]
Closing.
